File: src/main/java/io/github/hiiragi283/htms/api/material/HTMaterialFlags.java
package io.github.hiiragi283.htms.api.material;

import net.minecraft.util.ResourceLocation;

import org.jetbrains.annotations.NotNull;

import io.github.hiiragi283.htms.api.HTMaterialsAPI;

public final class HTMaterialFlags {

    private HTMaterialFlags() {}

    @NotNull
    public static final ResourceLocation GENERATE_DUST = HTMaterialsAPI.getId("generate_dust");

    @NotNull
    public static final ResourceLocation GENERATE_GEAR = HTMaterialsAPI.getId("generate_gear");

    @NotNull
    public static final ResourceLocation GENERATE_GEM = HTMaterialsAPI.getId("generate_gem");

    @NotNull
    public static final ResourceLocation GENERATE_INGOT = HTMaterialsAPI.getId("generate_ingot");

    @NotNull
    public static final ResourceLocation GENERATE_NUGGET = HTMaterialsAPI.getId("generate_nugget");

    @NotNull
    public static final ResourceLocation GENERATE_PLATE = HTMaterialsAPI.getId("generate_plate");

    @NotNull
    public static final ResourceLocation GENERATE_ROD = HTMaterialsAPI.getId("generate_rod");
}

File: src/main/kotlin/io/github/hiiragi283/api/material/property/HTMaterialFlags.kt
package io.github.hiiragi283.api.material.property

import io.github.hiiragi283.api.extension.TypedResourceLocation
import net.minecraft.util.ResourceLocation

object HTMaterialFlags {
    private val registry: MutableMap<ResourceLocation, TypedResourceLocation<Unit>> = hashMapOf()

    @JvmStatic
    fun getOrCreateFlag(id: ResourceLocation): TypedResourceLocation<Unit> =
        registry.computeIfAbsent(id, TypedResourceLocation.Companion::of)
}